File: scrape.py
import urllib.request
import re
import json
import ssl

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

req = urllib.request.Request("https://www.youtube.com/playlist?list=PLtJ-ufoDnA2DDcPhReJXU06cCF6KtE6eo", headers={"User-Agent": "Mozilla/5.0"})
html = urllib.request.urlopen(req, context=ctx).read().decode('utf-8')

ids = set()
vids = []

for match in re.finditer(r'\"title\":\{\"runs\":\[\{\"text\":\"(.*?)\"\}\].*?\"videoId\":\"([a-zA-Z0-9_-]{11})\"', html):
    t, v = match.group(1), match.group(2)
    t = t.replace('\\u0026', '&')
    if v not in ids and t != "Private video":
        ids.add(v)
        vids.append({"id": v, "title": t})

with open('vids.json', 'w', encoding='utf-8') as f:
    json.dump(vids, f)
print("Saved", len(vids))

File: scrape.js
const fs = require('fs');
const https = require('https');

https.get('https://www.youtube.com/playlist?list=PLtJ-ufoDnA2DDcPhReJXU06cCF6KtE6eo', { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
        const regex = /"title":{"runs":\[{"text":"(.*?)"}\]}.*?"videoId":"([a-zA-Z0-9_-]{11})"/g;
        let match;
        const seen = new Set();
        const vids = [];
        
        while ((match = regex.exec(data)) !== null) {
            const t = match[1].replace(/\\u0026/g, '&');
            const v = match[2];
            if (!seen.has(v) && t !== 'Private video') {
                seen.add(v);
                vids.append? vids.push({id: v, title: t}) : vids.push({id: v, title: t});
            }
        }
        
        fs.writeFileSync('vids.json', JSON.stringify(vids, null, 2));
        console.log('Saved', vids.length);
    });
}).on('error', err => console.log(err));
